src/AndcultureCode.CSharp.Sitefinity.Core/JsonSerialization/Converters/PageTypeConverter.cs

```diff
@@ -0,0 +1,29 @@
+﻿using AndcultureCode.CSharp.Sitefinity.Core.Enumerations;
+using Newtonsoft.Json;
+using System;
+
+namespace AndcultureCode.CSharp.Sitefinity.Core.JsonSerialization.Converters
+{
+    /// <summary>
+    /// Responsible for converting PageType enums values into numeric values
+    /// </summary>
+    public class PageTypeConverter : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            long convertedValue = ((PageType)value).GetHashCode();
+
+            writer.WriteValue(convertedValue.ToString("d"));
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            return reader.Value != null ? Enum.Parse(typeof(PageType), reader.Value.ToString()) : null;
```

Comment:
Should we consider `Enum.TryParse()` to appropriately handle errors when provided with invalid values?

Reply:
I'm not sure how much value we'd get from it -- we'd still want to throw an exception, and the error message returned by default from `Enum.Parse` I think is descriptive enough. E.g. I tested it on DotNetFiddle and got this: `System.ArgumentException: Requested value 'ThisWontBeFound' was not found.`

I believe `CanConvert` will also by default handle cases where we try to attach the converter to a property that isn't of type `PageType`.

Reply:
👍